### PR TITLE
Convert to go mod

### PR DESCRIPTION
**Pull request by wallyworld**
This PR converts the master branch to go mod v2.
Recent commits from gopkg.in/mgo.v2 (which fix some logging concurrency issues) were also pulled across.

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -278,7 +278,10 @@ NextTagSet:
 	return false
 }
 
-var pingDelay = 15 * time.Second
+var (
+	pingDelay   = 15 * time.Second
+	globalMutex sync.Mutex
```
Comment (hpidcock) [inline server.go:283]: Where is this even used now?

**Review by hpidcock — APPROVED** [review]
One question :question:

**Review by benhoyt — APPROVED** [review]
LGTM. There are a few places (eg `sim_test.go`) where the imports aren't consistently "stdlib group, 3rd party group". I'm not at all concerned about it -- just noting it as I noticed you fixed them up in some places.

File: server.go (hunk 1 above)
Comment (wallyworld) [reply server.go:283]: tests

**Comment by wallyworld** [comment]
> LGTM. There are a few places (eg `sim_test.go`) where the imports aren't consistently "stdlib group, 3rd party group". I'm not at all concerned about it -- just noting it as I noticed you fixed them up in some places.

Thanks, fixed some more

**Pull request merged**